libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -532,7 +537,7 @@ void AP_ADSB::send_dynamic_out(const mavlink_channel_t chan)
             ewVog,
             emStatus,
             state,
-            squawk);
+            (out_state.cfg.squawk & 0xFFFF));
```

Comment:
Aren't SQUAWK codes supposed to be 12 bit wide ?

Reply:
It is a 4 digit _octal_ code, which makes it 12 bit wide

Reply:
Yeah, that's right. I ended up having a call with Jeff Wheeler from uAvionix and got some more details. I need to add a few more additional things to this PR.